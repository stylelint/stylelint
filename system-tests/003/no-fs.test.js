'use strict';

const stylelint = require('../../lib');
const { caseConfig, caseCode, prepForSnapshot } = require('../systemTestUtils');

const CASE_NUMBER = '003';

it('no-fs - zen garden CSS with standard config', async () => {
	expect(
		prepForSnapshot(
			await stylelint.lint({
				code: await caseCode(CASE_NUMBER),
				config: await caseConfig(CASE_NUMBER),
				fix: true,
			}),
		),
	).toMatchSnapshot();
}, 10000);
