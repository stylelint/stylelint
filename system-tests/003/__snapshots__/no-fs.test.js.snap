// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no-fs - zen garden CSS with standard config 1`] = `
{
  "cwd": "",
  "errored": true,
  "output": "/* css Zen Garden default style v1.02 */

/* css released under Creative Commons License - http://creativecommons.org/licenses/by-nc-sa/1.0/  */

/* This file based on 'Tranquille' by Dave Shea */

/* You may use this file as a foundation for any new work, but you may find it easier to start from scratch. */

/* Not all elements are defined in this file, so you'll most likely want to refer to the xhtml as well. */

/* Your images should be linked as if the CSS file sits in the same folder as the images. ie. no paths. */

/* basic elements */
html {
  margin: 0;
  padding: 0;
}

body {
  font: 75% georgia, sans-serif;
  line-height: 1.88889;
  color: #555753;
  background: #fff url(http://csszengarden.com/001/blossoms.jpg) no-repeat bottom right;
  margin: 0;
  padding: 0;
}

p {
  margin-top: 0;
  text-align: justify;
}

h3 {
  font: italic normal 1.4em georgia, sans-serif;
  letter-spacing: 1px;
  margin-bottom: 0;
  color: #7d775c;
}

a:link {
  font-weight: bold;
  text-decoration: none;
  color: #b7a5df;
}

a:visited {
  font-weight: bold;
  text-decoration: none;
  color: #d4cddc;
}

a:hover,
a:focus,
a:active {
  text-decoration: underline;
  color: #9685ba;
}

abbr {
  border-bottom: none;
}

/* specific divs */
.page-wrapper {
  background: url(http://csszengarden.com/001/zen-bg.jpg) no-repeat top left;
  padding: 0 175px 0 110px;
  margin: 0;
  position: relative;
}

.intro {
  min-width: 470px;
  width: 100%;
}

header h1 {
  background: transparent url(http://csszengarden.com/001/h1.gif) no-repeat top left;
  margin-top: 10px;
  display: block;
  width: 219px;
  height: 87px;
  float: left;
  text-indent: 100%;
  white-space: nowrap;
  overflow: hidden;
}

header h2 {
  background: transparent url(http://csszengarden.com/001/h2.gif) no-repeat top left;
  margin-top: 58px;
  margin-bottom: 40px;
  width: 200px;
  height: 18px;
  float: right;
  text-indent: 100%;
  white-space: nowrap;
  overflow: hidden;
}

header {
  padding-top: 20px;
  height: 87px;
}

.summary {
  clear: both;
  margin: 20px 20px 20px 10px;
  width: 160px;
  float: left;
}

.summary p {
  font: italic 1.1em/2.2 georgia;
  text-align: center;
}

.preamble {
  clear: right;
  padding: 0 10px 0;
}

.supporting {
  padding-left: 10px;
  margin-bottom: 40px;
}

footer {
  text-align: center;
}

footer a:link,
footer a:visited {
  margin-right: 20px;
}

.sidebar {
  margin-left: 600px;
  position: absolute;
  top: 0;
  right: 0;
}

.sidebar .wrapper {
  font: 10px verdana, sans-serif;
  background: transparent url(http://csszengarden.com/001/paper-bg.jpg) top left repeat-y;
  padding: 10px;
  margin-top: 150px;
  width: 130px;
}

.sidebar h3.select {
  background: transparent url(http://csszengarden.com/001/h3.gif) no-repeat top left;
  margin: 10px 0 5px;
  width: 97px;
  height: 16px;
  text-indent: 100%;
  white-space: nowrap;
  overflow: hidden;
}

.sidebar h3.archives {
  background: transparent url(http://csszengarden.com/001/h5.gif) no-repeat top left;
  margin: 25px 0 5px;
  width: 57px;
  height: 14px;
  text-indent: 100%;
  white-space: nowrap;
  overflow: hidden;
}

.sidebar h3.resources {
  background: transparent url(http://csszengarden.com/001/h6.gif) no-repeat top left;
  margin: 25px 0 5px;
  width: 63px;
  height: 10px;
  text-indent: 100%;
  white-space: nowrap;
  overflow: hidden;
}

.sidebar ul {
  margin: 0;
  padding: 0;
}

.sidebar li {
  line-height: 1.3em;
  background: transparent url(http://csszengarden.com/001/cr1.gif) no-repeat top center;
  display: block;
  padding-top: 5px;
  margin-bottom: 5px;
  list-style-type: none;
}

.sidebar li a:link {
  color: #988f5e;
}

.sidebar li a:visited {
  color: #b3ae94;
}

.extra1 {
  background: transparent url(http://csszengarden.com/001/cr2.gif) top left no-repeat;
  position: absolute;
  top: 40px;
  right: 0;
  width: 148px;
  height: 110px;
}
",
  "reportedDisables": [],
  "results": [
    {
      "deprecations": [],
      "errored": true,
      "ignored": undefined,
      "invalidOptionWarnings": [],
      "parseErrors": [],
      "warnings": [
        {
          "column": 25,
          "endColumn": 32,
          "endLine": 102,
          "line": 102,
          "rule": "font-family-no-missing-generic-family-keyword",
          "severity": "error",
          "text": "Unexpected missing generic font family (font-family-no-missing-generic-family-keyword)",
        },
      ],
    },
  ],
  "ruleMetadata": {
    "at-rule-empty-line-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/at-rule-empty-line-before",
    },
    "at-rule-name-case": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/at-rule-name-case",
    },
    "at-rule-name-space-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/at-rule-name-space-after",
    },
    "at-rule-no-unknown": {
      "url": "https://stylelint.io/user-guide/rules/at-rule-no-unknown",
    },
    "at-rule-semicolon-newline-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/at-rule-semicolon-newline-after",
    },
    "block-closing-brace-empty-line-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/block-closing-brace-empty-line-before",
    },
    "block-closing-brace-newline-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/block-closing-brace-newline-after",
    },
    "block-closing-brace-newline-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/block-closing-brace-newline-before",
    },
    "block-closing-brace-space-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/block-closing-brace-space-before",
    },
    "block-no-empty": {
      "url": "https://stylelint.io/user-guide/rules/block-no-empty",
    },
    "block-opening-brace-newline-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/block-opening-brace-newline-after",
    },
    "block-opening-brace-space-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/block-opening-brace-space-after",
    },
    "block-opening-brace-space-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/block-opening-brace-space-before",
    },
    "color-hex-case": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/color-hex-case",
    },
    "color-hex-length": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/color-hex-length",
    },
    "color-no-invalid-hex": {
      "url": "https://stylelint.io/user-guide/rules/color-no-invalid-hex",
    },
    "comment-empty-line-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/comment-empty-line-before",
    },
    "comment-no-empty": {
      "url": "https://stylelint.io/user-guide/rules/comment-no-empty",
    },
    "comment-whitespace-inside": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/comment-whitespace-inside",
    },
    "custom-property-empty-line-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/custom-property-empty-line-before",
    },
    "declaration-bang-space-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/declaration-bang-space-after",
    },
    "declaration-bang-space-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/declaration-bang-space-before",
    },
    "declaration-block-no-duplicate-properties": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/declaration-block-no-duplicate-properties",
    },
    "declaration-block-no-shorthand-property-overrides": {
      "url": "https://stylelint.io/user-guide/rules/declaration-block-no-shorthand-property-overrides",
    },
    "declaration-block-semicolon-newline-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/declaration-block-semicolon-newline-after",
    },
    "declaration-block-semicolon-space-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/declaration-block-semicolon-space-after",
    },
    "declaration-block-semicolon-space-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/declaration-block-semicolon-space-before",
    },
    "declaration-block-single-line-max-declarations": {
      "url": "https://stylelint.io/user-guide/rules/declaration-block-single-line-max-declarations",
    },
    "declaration-block-trailing-semicolon": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/declaration-block-trailing-semicolon",
    },
    "declaration-colon-newline-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/declaration-colon-newline-after",
    },
    "declaration-colon-space-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/declaration-colon-space-after",
    },
    "declaration-colon-space-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/declaration-colon-space-before",
    },
    "declaration-empty-line-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/declaration-empty-line-before",
    },
    "font-family-no-duplicate-names": {
      "url": "https://stylelint.io/user-guide/rules/font-family-no-duplicate-names",
    },
    "font-family-no-missing-generic-family-keyword": {
      "url": "https://stylelint.io/user-guide/rules/font-family-no-missing-generic-family-keyword",
    },
    "function-calc-no-unspaced-operator": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/function-calc-no-unspaced-operator",
    },
    "function-comma-newline-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/function-comma-newline-after",
    },
    "function-comma-space-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/function-comma-space-after",
    },
    "function-comma-space-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/function-comma-space-before",
    },
    "function-linear-gradient-no-nonstandard-direction": {
      "url": "https://stylelint.io/user-guide/rules/function-linear-gradient-no-nonstandard-direction",
    },
    "function-max-empty-lines": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/function-max-empty-lines",
    },
    "function-name-case": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/function-name-case",
    },
    "function-parentheses-newline-inside": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/function-parentheses-newline-inside",
    },
    "function-parentheses-space-inside": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/function-parentheses-space-inside",
    },
    "function-whitespace-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/function-whitespace-after",
    },
    "indentation": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/indentation",
    },
    "keyframe-declaration-no-important": {
      "url": "https://stylelint.io/user-guide/rules/keyframe-declaration-no-important",
    },
    "length-zero-no-unit": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/length-zero-no-unit",
    },
    "max-empty-lines": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/max-empty-lines",
    },
    "media-feature-colon-space-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/media-feature-colon-space-after",
    },
    "media-feature-colon-space-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/media-feature-colon-space-before",
    },
    "media-feature-name-case": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/media-feature-name-case",
    },
    "media-feature-name-no-unknown": {
      "url": "https://stylelint.io/user-guide/rules/media-feature-name-no-unknown",
    },
    "media-feature-parentheses-space-inside": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/media-feature-parentheses-space-inside",
    },
    "media-feature-range-operator-space-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/media-feature-range-operator-space-after",
    },
    "media-feature-range-operator-space-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/media-feature-range-operator-space-before",
    },
    "media-query-list-comma-newline-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/media-query-list-comma-newline-after",
    },
    "media-query-list-comma-space-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/media-query-list-comma-space-after",
    },
    "media-query-list-comma-space-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/media-query-list-comma-space-before",
    },
    "no-descending-specificity": {
      "url": "https://stylelint.io/user-guide/rules/no-descending-specificity",
    },
    "no-duplicate-at-import-rules": {
      "url": "https://stylelint.io/user-guide/rules/no-duplicate-at-import-rules",
    },
    "no-duplicate-selectors": {
      "url": "https://stylelint.io/user-guide/rules/no-duplicate-selectors",
    },
    "no-empty-source": {
      "url": "https://stylelint.io/user-guide/rules/no-empty-source",
    },
    "no-eol-whitespace": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/no-eol-whitespace",
    },
    "no-extra-semicolons": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/no-extra-semicolons",
    },
    "no-invalid-double-slash-comments": {
      "url": "https://stylelint.io/user-guide/rules/no-invalid-double-slash-comments",
    },
    "no-missing-end-of-source-newline": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/no-missing-end-of-source-newline",
    },
    "number-leading-zero": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/number-leading-zero",
    },
    "number-no-trailing-zeros": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/number-no-trailing-zeros",
    },
    "property-case": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/property-case",
    },
    "property-no-unknown": {
      "url": "https://stylelint.io/user-guide/rules/property-no-unknown",
    },
    "rule-empty-line-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/rule-empty-line-before",
    },
    "selector-attribute-brackets-space-inside": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/selector-attribute-brackets-space-inside",
    },
    "selector-attribute-operator-space-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/selector-attribute-operator-space-after",
    },
    "selector-attribute-operator-space-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/selector-attribute-operator-space-before",
    },
    "selector-combinator-space-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/selector-combinator-space-after",
    },
    "selector-combinator-space-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/selector-combinator-space-before",
    },
    "selector-descendant-combinator-no-non-space": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/selector-descendant-combinator-no-non-space",
    },
    "selector-list-comma-newline-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/selector-list-comma-newline-after",
    },
    "selector-list-comma-space-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/selector-list-comma-space-before",
    },
    "selector-max-empty-lines": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/selector-max-empty-lines",
    },
    "selector-pseudo-class-case": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/selector-pseudo-class-case",
    },
    "selector-pseudo-class-no-unknown": {
      "url": "https://stylelint.io/user-guide/rules/selector-pseudo-class-no-unknown",
    },
    "selector-pseudo-class-parentheses-space-inside": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/selector-pseudo-class-parentheses-space-inside",
    },
    "selector-pseudo-element-case": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/selector-pseudo-element-case",
    },
    "selector-pseudo-element-colon-notation": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/selector-pseudo-element-colon-notation",
    },
    "selector-pseudo-element-no-unknown": {
      "url": "https://stylelint.io/user-guide/rules/selector-pseudo-element-no-unknown",
    },
    "selector-type-case": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/selector-type-case",
    },
    "selector-type-no-unknown": {
      "url": "https://stylelint.io/user-guide/rules/selector-type-no-unknown",
    },
    "string-no-newline": {
      "url": "https://stylelint.io/user-guide/rules/string-no-newline",
    },
    "unit-case": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/unit-case",
    },
    "unit-no-unknown": {
      "url": "https://stylelint.io/user-guide/rules/unit-no-unknown",
    },
    "value-keyword-case": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/value-keyword-case",
    },
    "value-list-comma-newline-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/value-list-comma-newline-after",
    },
    "value-list-comma-space-after": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/value-list-comma-space-after",
    },
    "value-list-comma-space-before": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/value-list-comma-space-before",
    },
    "value-list-max-empty-lines": {
      "fixable": true,
      "url": "https://stylelint.io/user-guide/rules/value-list-max-empty-lines",
    },
  },
}
`;
